boot/bootutil/include/bootutil/boot_status.h

```diff
@@ -119,6 +119,7 @@ extern "C" {
 #define BLINFO_MAX_APPLICATION_SIZE_IMAGE_2 0x07
 #define BLINFO_MAX_APPLICATION_SIZE_IMAGE_3 0x08
 #define BLINFO_MAX_APPLICATION_SIZE_IMAGE_4 0x09
+#define BLINFO_FLASH_AREA           0x0A
```

Comment:
Isn't this already done vis BLINFI_RUNNING_SLOT?

Reply:
`RUNNING_SLOT` gives the index of the active slot, but for the MCUBoot subsystem code in Zephyr we need access to the flash area ID. For example, Slot 0 on the RT1050 has flash area ID 1 (because the MCUBoot partition has area ID 0)

We could change the value set by `RUNNING_SLOT`, but I wanted to avoid breaking any APIs so I added this. It was needed in order to get features like confirming a running image working with MCUBoot

Reply:
You have to convert from running slot to flash area ID in your application itself, as img mgmt does e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c#L128 the flash area ID has no meaning between different applications e.g. application could have bootloader slot split into 2 in future, now you open the wrong thing

Reply:
Ahh, ok. I was testing this with the DFU subsystem- so would it make sense to call `img_mgmt_flash_area_id` here in `boot_fetch_active_slot`: https://github.com/zephyrproject-rtos/zephyr/pull/85254/files#diff-696b0049cfb981ee66f0191ccaf755f25a632c35b86da7dc1d739c3de8a60ed3L78? Otherwise, functions like marking the image as pending weren't working, since `boot_fetch_active_slot` is really fetching the flash area ID of the active slot, not the slot number.

Reply:
That returns either:
```
#define BOOT_PRIMARY_SLOT               0
#define BOOT_SECONDARY_SLOT             1
```
Which should then be used to check if it's slot0 if 0 or slot1 if 1 (it doesn't care about additional images since they are not "running"),